base/strings/search.jl

```diff
@@ -166,35 +188,40 @@ function _search_bloom_mask(c)
 end
 
 _nthbyte(s::String, i) = codeunit(s, i)
-_nthbyte(a::Union{AbstractVector{UInt8},AbstractVector{Int8}}, i) = a[i]
+_nthbyte(a::Union{Vector{UInt8},Vector{Int8}}, i) = a[i]
+_nthbyte(t::AbstractVector, index) = t[firstindex(t) + (index-1)]
 
 function _searchindex(s::String, t::String, i::Integer)
     # Check for fast case of a single byte
     lastindex(t) == 1 && return something(findnext(isequal(t[1]), s, i), 0)
     _searchindex(unsafe_wrap(Vector{UInt8},s), unsafe_wrap(Vector{UInt8},t), i)
 end
 
-function _searchindex(s::ByteArray, t::ByteArray, i::Integer)
-    n = sizeof(t)
-    m = sizeof(s)
+function _searchindex(s::AbstractVector{<:Union{Int8,UInt8}},
+                      t::AbstractVector{<:Union{Int8,UInt8}},
+                      i::Integer)
+    n = length(t)
+    m = length(s)
+    f_s = firstindex(s)
+    i < f_s && throw(BoundsError(s, i))
```

Comment:
Since you are doing 1-based index calculations below, you should shift `i` to a 1-based index.    I would declare the parameter as `_i::Integer`, and then do:
```jl
i = Int(_i) - (firstindex(s) - 1)
i < 1 && throw(BoundsError(s, _i))
```
to shift `i` to a `1`-based index for use with `nthbyte` and your other 1-based calculations.